### PR TITLE
gh-102595: Document `PyObject_Format` c-api function

### PR DESCRIPTION
**Pull request by sobolevn**
Def: https://github.com/python/cpython/blame/5ffdaf748d98da6065158534720f1996a45a0072/Include/abstract.h#L389

* Issue: gh-102595

Automerge-Triggered-By: GH:encukou

### CODE REVIEW

**Comment by sobolevn** [comment]
@JelleZijlstra @AlexWaygood I don't know who else might take a look, github does not suggest anybody :)
Would you, please?

File: Doc/c-api/object.rst [hunk 1]
```diff
@@ -179,6 +179,13 @@ Object Protocol
    If *o1* and *o2* are the same object, :c:func:`PyObject_RichCompareBool`
    will always return ``1`` for :const:`Py_EQ` and ``0`` for :const:`Py_NE`.
 
+.. c:function:: PyObject* PyObject_Format(PyObject *obj, PyObject *format_spec)
```
Comment (JelleZijlstra) [inline Doc/c-api/object.rst:182]: Weird that we're inconsistent in putting the space before or after the `*`, but the next two functions do the same thing.

File: Doc/c-api/object.rst [hunk 2]
```diff
@@ -179,6 +179,13 @@ Object Protocol
...
+
+   Takes an arbitrary *obj* and returns the result of calling
+   ``obj.__format__(format_spec)``.
+
+   *format_spec* might be ``NULL`` when ``format(obj)`` is used.
```
Comment (JelleZijlstra) [inline Doc/c-api/object.rst]:
```suggestion
   Format *obj* using *format_spec*. This is equivalent to the Python
   expression ``format(obj, format_spec)``.

   *format_spec* may be ``NULL``. In this case the call is equivalent
   to ``format(obj)``.
   
   Returns the formatted string on success, ``NULL`` on failure.
```
- Closer in wording to repr() and str() below
- Mention the return type

**Review by JelleZijlstra — COMMENTED** [review]
I can review the text, but going to request review from @encukou to confirm whether this should be documented, as I am not an expert on the C API.

**Comment by AlexWaygood** [comment]
I'm also not an expert on the C API, and also would have suggested @encukou as a reviewer for this one (though his GitHub status says he's ill right now 🙁). With @JelleZijlstra's suggestions, the wording looks good, though!

**Comment by JelleZijlstra** [comment]
> (though his GitHub status says he's ill right now 🙁).

I didn't notice that, sorry for pinging you Petr!

**Comment by sobolevn** [comment]
Get well soon, Petr! :)

**Review by JelleZijlstra — APPROVED** [review]

File: Doc/c-api/object.rst [hunk 3]
```diff
@@ -179,6 +179,16 @@ Object Protocol
...
    will always return ``1`` for :const:`Py_EQ` and ``0`` for :const:`Py_NE`.
 
+.. c:function:: PyObject* PyObject_Format(PyObject *obj, PyObject *format_spec)
+
+   Format *obj* using *format_spec*. This is equivalent to the Python
+   expression ``format(obj, format_spec)``.
+
+   *format_spec* may be ``NULL``. In this case the call is equivalent
+   to ``format(obj)``.
+
```
Comment (JelleZijlstra) [inline Doc/c-api/object.rst]:
```suggestion

```

File: Doc/c-api/object.rst (hunk 1 above)
Comment (encukou) [reply Doc/c-api/object.rst:182]: The relevant style guide would be PEP 7, but it says nothing about this issue. All the examples use `PyObject *x`, though.

**Comment by encukou** [comment]
Thanks for the wishes, and for the docs!
IMO, this definitely should be documented.

**Review by encukou — APPROVED** [review]

**Pull request merged**